File: Objective-C Example/SmoothCorners/ViewController.m
//
//  ViewController.m
//  SmoothCorners
//
//  Created by Felix Lapalme on 2019-03-16.
//  Copyright © 2019 Felix Lapalme. All rights reserved.
//

#import "ViewController.h"
#import "UIView+SmoothCorners.h"
#import "FLXSmoothView.h"

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];

    UIView *redView = [UIView new];
    UIView *yellowView = [UIView new];
    FLXSmoothView *blueView = [FLXSmoothView new];

    redView.backgroundColor = [[UIColor redColor] colorWithAlphaComponent:0.9];
    yellowView.backgroundColor = [[UIColor yellowColor] colorWithAlphaComponent:0.9];
    blueView.backgroundColor = [[UIColor blueColor] colorWithAlphaComponent:0.4];

    redView.frame = CGRectInset(self.view.bounds, 20, 50);
    yellowView.frame = redView.frame;
    blueView.frame = redView.frame;

    [self.view addSubview:redView];
    [self.view addSubview:yellowView];
    [self.view addSubview:blueView];

    CGFloat radius = 90;

    // red view with regular corners
    redView.layer.cornerRadius = radius;

    // yellow view with smooth corners (using UIView category)
    yellowView.flx_continuousCorners = YES;
    yellowView.layer.cornerRadius = radius;

    // blue view (UIView subclass)
    blueView.flx_smoothCorners = YES;
    blueView.layer.cornerRadius = radius;

    UILabel *explanationLabel = [UILabel new];
    explanationLabel.frame = yellowView.frame;
    explanationLabel.numberOfLines = 0;
    explanationLabel.text = @"Red = regular rounded corners\nBeige = smooth rounded corners";
    [self.view addSubview:explanationLabel];
}

@end

File: Swift Example/SmoothCorners/ViewController.swift
//
//  ViewController.swift
//  SmoothCorners
//
//  Created by Felix Lapalme on 2019-03-17.
//  Copyright © 2019 Felix Lapalme. All rights reserved.
//

import UIKit

class ViewController: UIViewController {

    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view.
        let redView = UIView()
        let yellowView = UIView()
        let blueView = SmoothView()

        redView.backgroundColor = UIColor.red.withAlphaComponent(0.9)
        yellowView.backgroundColor = UIColor.yellow.withAlphaComponent(0.9)
        blueView.backgroundColor = UIColor.blue.withAlphaComponent(0.4)

        redView.frame = view.bounds.insetBy(dx: 20, dy: 50)
        yellowView.frame = redView.frame
        blueView.frame = redView.frame

        view.addSubview(redView)
        view.addSubview(yellowView)
        view.addSubview(blueView)

        let radius : CGFloat = 90.0

        // red view with regular corners
        redView.layer.cornerRadius = radius

        // using category/extension
        yellowView.flx_continuousCorners = true
        yellowView.layer.cornerRadius = radius

        // using subclass
        blueView.flx_smoothCorners = true
        blueView.layer.cornerRadius = radius

        let explanationLabel = UILabel()
        explanationLabel.frame = yellowView.frame
        explanationLabel.numberOfLines = 0
        explanationLabel.text = "Red = regular rounded corners\nBeige = smooth rounded corners"
        view.addSubview(explanationLabel)
    }
}

